code/components/discord/src/DiscordPresence.cpp

```diff
@@ -108,4 +158,40 @@ static InitFunction initFunction([]()
 
 		g_richPresenceChanged = true;
 	});
+
+	fx::ScriptEngine::RegisterNativeHandler("SET_RICH_PRESENCE_ASSET", [](fx::ScriptContext& context)
```

Comment:
Service-specific, so make the native like `SET_DISCORD_RICH_PRESENCE_ASSET`. Also, you forgot to update `codegen_cfx_natives.lua`?